### PR TITLE
chore(storage): add default gRPC connection pool

### PR DESCRIPTION
**Pull request by noahdietz**
Adds a default gRPC connection pool of 4 for the gRPC client library. This is the same default used by BigTable, BigQuery Storage, and PubSub.

### CODE REVIEW

File: storage/storage.go [hunk 1]
```diff
@@ -81,6 +81,10 @@ const (
 	// ScopeReadWrite grants permissions to manage your
 	// data in Google Cloud Storage.
 	ScopeReadWrite = raw.DevstorageReadWriteScope
+
+	// defualtConnPoolSize is the default number of connections
```
Comment (crwilcox) [inline storage/storage.go:85]: nit: typo

Comment (noahdietz) [reply storage/storage.go]: Done! Thanks :)

File: storage/storage.go [hunk 2]
```diff
@@ -81,6 +81,10 @@ const (
 	// ScopeReadWrite grants permissions to manage your
 	// data in Google Cloud Storage.
 	ScopeReadWrite = raw.DevstorageReadWriteScope
+
+	// defaultConnPoolSize is the default number of connections
+	// to initialize in the GAPIC gRPC connection pool.
```
Comment (tritone) [inline storage/storage.go:86]: Maybe for later but do we want to add a comment about how a larger pool may be required for high-throughput operations?

**Review by tritone — COMMENTED** [review]
Generally looks good!

File: storage/storage.go (hunk 2 above)
Comment (noahdietz) [reply storage/storage.go:86]: I will add something here, but the proper documentation I think should be written when we take this public.

Comment (noahdietz) [reply storage/storage.go]: Added a bit.

**Review by tritone — APPROVED** [review]

**Pull request merged**